### ISSUE TITLE
Error: Negative dimension size caused by subtracting 3 from 1

### ISSUE THREAD

**Issue by Sorena1994** [keras-team/keras#12951]
First of all, I read the previous OPs, and none of them worked. 
Here's the error:  Negative dimension size caused by subtracting 3 from 1 for 'conv2d_8/convolution' (op: 'Conv2D') with input shapes: [?,1,1,64], [3,3,64,128]
Here's the code: 
```
model = models.Sequential()
model.add(layers.Conv2D(16, (3, 3), activation='relu', input_shape=(28,28, 1)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(32, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(128, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))

```
I would appreciate to know how can I fix the error, and generally what these type of errors mean. 
Thanks everyone.

**Comment by sourav22899** [comment]
The default `padding` is `valid` in `keras.layers.Conv2D`. Using `valid` padding reduces dimensions after each convolution layer. Use `padding='same'` like:
`model.add(layers.Conv2D(16, (3, 3), activation='relu', input_shape=(28,28, 1),padding='same'))
`

**Comment by sourav22899** [comment]
Please close the issue if the solution serves your purpose.

Best,
Sourav
